jmx/src/main/java/org/jboss/as/jmx/model/MBeanInfoFactory.java

```diff
@@ -263,9 +264,12 @@ private OpenMBeanParameterInfo[] getParameterInfos(OperationDefinition opDef, Mo
             boolean expressionsAllowed = value.hasDefined(EXPRESSIONS_ALLOWED) && value.get(EXPRESSIONS_ALLOWED).asBoolean();
             descriptions.put(DESC_EXPRESSIONS_ALLOWED, String.valueOf(expressionsAllowed));
 
+            OpenType<?> mbeanType = converters.convertToMBeanType(attributeDefinition, value);
             if (!expressionsAllowed) {
                 Object defaultValue = getIfExists(attributeDefinition, value, DEFAULT);
-                descriptions.put(DEFAULT_VALUE_FIELD, defaultValue);
+                if (!mbeanType.isArray() && !((Object)mbeanType instanceof TabularType) ) {
+                    descriptions.put(DEFAULT_VALUE_FIELD, defaultValue);
+                }
```

Comment:
Perhaps add a comment? AIUI this is basically a workaround, which is fine, but commenting that it's a workaround may help future maintainers who otherwise may go into a rabbit hole trying to understand.